### ISSUE TITLE
Serializing ID with JsonProperty doesn't work 

### ISSUE THREAD

**Issue by leolorenzoluis** [Azure/azure-webjobs-sdk#1214]
Please provide a succinct description of the issue.

#### Repro steps

Provide the steps required to reproduce the problem

1. Create the following class:
```
Given a POJO with the ff:
```
 public class ProfilePictureUrl 
    {
        public string UserId { get; set; }

        public string Url { get; set; }

        [JsonProperty(PropertyName = "id")]
        public string Id { get; set; }
    }
```

2. When the following method is called, it still expects a `id` property, so that means `JsonProperty` did not serialize it correctly. However, if I change the property name to `id` then it works fine.
```
 [FunctionName(nameof(ResizeImage))]
        public static async Task RunAsync([BlobTrigger("profile-pictures/{name}")] CloudBlockBlob myBlob, string name, [DocumentDB(databaseName: "DocumentDB", collectionName: "ResizedProfilePictures", CreateIfNotExists = true)] DocumentClient client, [Blob("resized-profile-pictures/{name}", FileAccess.ReadWrite)] CloudBlockBlob resizedBlob, TraceWriter log)
        {
            resizedBlob.Properties.ContentType = myBlob.Properties.ContentType;
            using (var stream = new MemoryStream())
            {
                await myBlob.DownloadToStreamAsync(stream);
                stream.Position = 0;

                log.Info($"C# Blob trigger function Processed blob\n Name:{name} \n Size: {stream.Length} Bytes");
                SendThumbnailRequest(stream, log, resizedBlob);

                var exists = client.CreateDocumentQuery<ProfilePictureUrl>(
                    UriFactory.CreateDocumentCollectionUri("DocumentDB", "ResizedProfilePictures"),
                    new FeedOptions() { MaxItemCount = -1 }).Where(x => x.UserId == myBlob.Metadata["UserId"]).AsEnumerable().FirstOrDefault();
                if (exists != null)
                {
                    exists.Url = resizedBlob.Uri.AbsoluteUri;
                    await client.ReplaceDocumentAsync(
                    UriFactory.CreateDocumentUri("Facebook", "ResizedProfilePictures", exists.Id), exists);
                }
                else
                {
                    await client.CreateDocumentAsync(UriFactory.CreateDocumentCollectionUri("DocumentDB", "ResizedProfilePictures"), new ProfilePictureUrl() { Url = resizedBlob.Uri.AbsoluteUri, UserId = myBlob.Metadata["UserId"] });
                }
            }
        }
```

#### Expected behavior

It should successfully update the document.

#### Actual behavior

mscorlib: Exception while executing function: Functions.ResizeImage. Microsoft.Azure.Documents.Client: Message: {"Errors":["The input content is invalid because the required properties - 'id; ' - are missing","The request payload is invalid. Ensure to provide a valid request payload."]}

#### Known workarounds

Provide a description of any known workarounds.

#### Related information 

Provide any related information 

* Azure.DocumentDbExtensions 1.1.0-beta1

**Comment by christopheranderson** [comment]
Might be multiple issues here.

Are you intentionally not setting id here?

```csharp
await client.CreateDocumentAsync(UriFactory.CreateDocumentCollectionUri("DocumentDB", "ResizedProfilePictures"), new ProfilePictureUrl() { Url = resizedBlob.Uri.AbsoluteUri, UserId = myBlob.Metadata["UserId"] });
```

Are you seeing this on your update workflow as well?

**Comment by leolorenzoluis** [comment]
I am setting the id property. 

I'm not seeing this on my update workflow. This is running through the local Azure Function CLI.

I am using the latest version of Json which I believe is 10.x.

**Comment by enihar** [comment]
@leolorenzoluis  Did you manage to find a solution? I too have the same issue.

**Comment by formix** [comment]
Same issue here but when returning an instance of a type. Two of my properties have Json attributes set (JsonIgnore and JsonProperty) and neither are taken into account by the serializer.

I don't think this issue is related to a property name (ID) bu rather that **Azure Functions** do not use Newtonsoft.Json at all for serialization. If we can fin what is the underlying serializer, there is a good chance that we'll find how to control serialization (if it provides that kind of hook).

**Comment by formix** [comment]
Indeed, the **Azure Functions** use a `DataContractJsonSerializer` which is an [MS class](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.json.datacontractjsonserializer?view=netframework-4.7.2). To make work with Newtonsoft.Json, you have to follow that [stackoverflow](https://stackoverflow.com/questions/48638174/http-triggers-in-azure-functions-serializing-objects-in-camelcase) selected answer. 

Instead of a `CamelCasePropertyNamesContractResolver`, provide a `DefaultContractResolver` and it should work.

I tried that for my case and it threw a "Method not found" exception because I have another dependency that requires me to link to Newtonsoft.Json >=10.0.1 while the Azure Function dependency on N.Json == 9.0.1. and in later version the method do not exist.

Hope it helps!

**Comment by formix** [comment]
Ok, after some digging, I recalled that `DataContractJsonSerializer` from System.Serialization when I used to do web services with WCF. Using the `System.Runtime.Serialization`. You should be able to control the serializer behavior without having to deal with Newtonsoft.Json.

Use the `DataMemberAttribute` to set a different name to a class property. Other attributes exist as well to deal with other serialization cases like IgnoreDataMemberAttribute, etc.

Confirmed! Using `System.Runtime.Serialization` fixed the issue!
